charts/ibm-fhir-server/templates/deployment.yaml

```
@@ -123,44 +140,32 @@ spec:
               value: "-1"
             {{- if .Values.objectStorage.enabled }}
             - name: BATCH_DB_HOSTNAME
-              value: "{{ .Values.db.host }}"
+              value: "{{ include "fhir.database.host" $ }}"
             - name: BATCH_DB_PORT
-              value: "{{ .Values.db.port }}"
+              value: "{{ include "fhir.database.port" $ }}"
             - name: BATCH_DB_NAME
-              value: "{{ .Values.db.name }}"
+              value: "{{ include "fhir.database.name" $ }}"
             - name: BATCH_DB_USER
-              value: "{{ .Values.db.user }}"
-            {{- if .Values.db.dbSecret }}
-            {{- if .Values.db.passwordSecretKey }}
+              value: "{{ include "fhir.database.user" $ }}"
             - name: BATCH_DB_PASSWORD
               valueFrom:
                 secretKeyRef:
-                  name: {{ .Values.db.dbSecret }}
-                  key: {{ .Values.db.passwordSecretKey }}
-            {{- end }}
+                  name: "{{ include "fhir.database.secretName" $ }}"
+                  key: "{{ include "fhir.database.secretKey" $ }}"
```

```suggestion
                  key: "{{ include "fhir.database.passwordSecretKey" $ }}"
```
per comment in _helpers.tpl

---

lines 155-168 can be replaced with the following ((based on comment in _helpers.tpl):
```
            - name: BATCH_DB_APIKEY
              valueFrom:
                secretKeyRef:
                  name: "{{ include "fhir.database.secretName" $ }}"
                  key: "{{ include "fhir.database.apiKeySecretKey" $ }}"
```